### DISCUSSION TITLE
Integration Discussion

### DISCUSSION THREAD

**Discussion by jkleiber** [team401/2024-Robot-Code#120] [General]
How did things go today?

**Comment by honzikschenk** [comment]
@jkleiber Coincidentally, I was just about to make this post updating you. :)

### Thursday (Feb. 22) Update
The robot works!
- Auto-align does some weird stuff where it jumps between positions (I believe that this may be due to weird velocities being calculated).
- We tried intaking and shooting from different positions and the both work surprisingly well with the limited data we gave it yesterday.
- The banner sensor is really inconsistent/laggy so we may have to mess with the index speeds a little.
- Endgame positions work (we don't have hood yet)
- Stage avoidance also works as intended

We got a [video](https://photos.app.goo.gl/vmkee49zydCvdr3B8) of the endgame but not the scoring unfortunately.

**Reply by jkleiber** [reply]
That's great! So I assume we have the following items to resolve before we start working on auto:

- [x] Get the BeeLink and Photonvision set up and integrated into the navigation system
- [x] Fix the auto aim weird velocity thing
- [x] Get the belt/kicker speeds right during intake
- [x] Tune drivetrain constants to be better

Then we also have these things we need to do in general:
- [x] tune the hood? And test if we can soft toss into the amp
- [ ] test/tune Aiden's climb controller

Feel free to add anything else if I missed anything

**Comment by jkleiber** [comment]
**Sunday Planning**
I might actually be able to make it for part of shop today, but regardless we need to do the following:

- [x] Test the brake mode switch / maintenance timer (this should be fast), merge if possible 
- [ ] Test the JSON bindings, merge if possible 
- [ ] Test and tune our climb controller, merge if possible 
- [x] Characterize drivetrain feedforwards (likely new PR)
- [x] Get vision working (potentially a new PR depending on how that goes)

These are all things we can do without the intake/uptake if that needs to be repaired or something. We can also drive the auto paths without intaking after all these things in order to tune our path following controller.

cc: @redPlover @charlesmackenzie @PChild
